tests/testsuite/check_cfg.rs

```diff
@@ -0,0 +1,463 @@
+    if !is_nightly() {
+        // --check-cfg is a nightly only rustc command line
+        return;
+    }
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [project]
+                name = "foo"
+                version = "0.1.0"
+
+                [dependencies]
+                bar = { path = "bar/", optional = true }
+
+                [features]
+                default = ["bar"]
+                f_a = []
+                f_b = []
+            "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .file("bar/Cargo.toml", &basic_manifest("bar", "0.1.0"))
+        .file("bar/src/lib.rs", "#[allow(dead_code)] fn bar() {}")
+        .build();
+
+    p.cargo("build -v -Zcheck-cfg=features")
+        .masquerade_as_nightly_cargo()
+        .with_stderr_contains(x!("rustc" => "values" of "feature"))
+        .with_stderr_contains(x!("rustc" => "values" of "feature" with "bar" "default" "f_a" "f_b"))
+        .run();
+}
+
+#[cargo_test]
+fn features_with_namespaced_features() {
+    if !is_nightly() {
+        // --check-cfg is a nightly only rustc command line
+        return;
+    }
+
+    let p = project()
+        .file(
+            "Cargo.toml",
+            r#"
+                [project]
+                name = "foo"
+                version = "0.1.0"
+
+                [dependencies]
+                bar = { path = "bar/", optional = true }
+
+                [features]
+                f_a = ["dep:bar"]
+                f_b = []
+            "#,
+        )
+        .file("src/main.rs", "fn main() {}")
+        .file("bar/Cargo.toml", &basic_manifest("bar", "0.1.0"))
+        .file("bar/src/lib.rs", "#[allow(dead_code)] fn bar() {}")
+        .build();
+
+    p.cargo("build -v -Zcheck-cfg=features")
+        .masquerade_as_nightly_cargo()
+        .with_stderr_contains(x!("rustc" => "values" of "feature" with "f_a" "f_b"))
+        .run();
+}
+
+#[cargo_test]
+fn well_known_names() {
+    if !is_nightly() {
+        // --check-cfg is a nightly only rustc command line
+        return;
+    }
+
+    let p = project()
+        .file("Cargo.toml", &basic_manifest("foo", "0.1.0"))
+        .file("src/main.rs", "fn main() {}")
+        .build();
+
+    p.cargo("build -v -Zcheck-cfg=names")
+        .masquerade_as_nightly_cargo()
+        .with_stderr_contains(x!("rustc" => "names"))
+        .run();
+}
+
+#[cargo_test]
+fn well_known_values() {
+    if !is_nightly() {
+        // --check-cfg is a nightly only rustc command line
+        return;
+    }
+
+    let p = project()
+        .file("Cargo.toml", &basic_manifest("foo", "0.1.0"))
+        .file("src/main.rs", "fn main() {}")
+        .build();
+
+    p.cargo("build -v -Zcheck-cfg=values")
+        .masquerade_as_nightly_cargo()
+        .with_stderr_contains(x!("rustc" => "values"))
+        .run();
+}
+
+#[cfg_attr(windows, ignore)] // weird normalization issue with windows and cargo-test-support
```

Comment:
The `cli_all_options` test didn't seem to be updated for windows, was that intentional?

Reply:
Yes it was, because I didn't wanted have to complicate the `x!` to be able to handle multi match on the same invocation, but your comment made me realize that I could simply check them independently, which I just did.